core/adapters/condition.go

```diff
@@ -0,0 +1,104 @@
+package adapters
+
+import (
+	"errors"
+	"strconv"
+
+	"github.com/smartcontractkit/chainlink/core/store"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"github.com/tidwall/gjson"
+)
+
+// Condition adapter type takes an Operator and a Value field to
+// compare to the previous task's Result.
+type Condition struct {
+	Operator string `json:"operator"`
+	Value    string `json:"value"`
+}
+
+var (
+	ErrResultNotNumber      = errors.New("The result was not a number")
+	ErrValueNotNumber       = errors.New("The value was not a number")
+	ErrOperatorNotSpecified = errors.New("Operator not specified")
+	ErrValueNotSpecified    = errors.New("Value not specified")
+)
+
+// Perform uses the Operator to check the run's result against the
+// specified Value.
+func (c *Condition) Perform(input models.RunResult, _ *store.Store) models.RunResult {
+	prevResult := input.Result()
+
+	if c.Value == "" {
+		input.SetError(ErrValueNotSpecified)
+		return input
+	}
+
+	switch c.Operator {
+	case "eq":
+		if c.Value == prevResult.String() {
+			input.CompleteWithResult(true)
+		} else {
+			input.CompleteWithResult(false)
+		}
+	case "gt":
+		value, desired, err := getValues(prevResult, c.Value)
+		if err != nil {
+			input.SetError(err)
+			return input
+		}
+		if desired < value {
+			input.CompleteWithResult(true)
+		} else {
+			input.CompleteWithResult(false)
+		}
+	case "gte":
+		value, desired, err := getValues(prevResult, c.Value)
+		if err != nil {
+			input.SetError(err)
+			return input
+		}
+		if desired <= value {
+			input.CompleteWithResult(true)
+		} else {
+			input.CompleteWithResult(false)
+		}
+	case "lt":
+		value, desired, err := getValues(prevResult, c.Value)
+		if err != nil {
+			input.SetError(err)
+			return input
+		}
+		if desired > value {
+			input.CompleteWithResult(true)
+		} else {
+			input.CompleteWithResult(false)
+		}
+	case "lte":
+		value, desired, err := getValues(prevResult, c.Value)
+		if err != nil {
+			input.SetError(err)
+			return input
+		}
+		if desired >= value {
+			input.CompleteWithResult(true)
+		} else {
+			input.CompleteWithResult(false)
+		}
+	default:
+		input.SetError(ErrOperatorNotSpecified)
+	}
+
+	return input
+}
+
+func getValues(result gjson.Result, d string) (float64, float64, error) {
+	value, err := strconv.ParseFloat(result.String(), 64)
```

Comment:
Is this the same type as we assume numbers are in the rest of our tasks? I think we use ethuint256 in other places. Also I would expect this to work with strings/objects etc.

Reply:
I'm not completely following you. `ethuint256` is another adapter that formats the result to an Ethereum data type. This adapter would be dealing with the result from an API, before any formatting for the blockchain. So you can think of a task list to look like:
```json
"tasks": [
  {
    "type": "httpget", "params": {"get":"http://example.com/api"}
  },
  {
    "type": "jsonparse", "params": {"path":["some","path"]}
  },
  {
    "type": "condition", "params": {"operator":"eq","value":"42"}
  },
  {
    "type": "ethbool"
  },
  {
    "type": "ethtx"
  }
]
```
So we're comparing the result of a field from the API to a value for the condition check. If the result at `some.path` is 42, we return `true`, or `false` otherwise.

> Also I would expect this to work with strings/objects etc.

For simplicity's sake, I'm not comparing entire objects, just a single field.

Reply:
One could conceivably do this:

```
"tasks": [
...
 {
  "type": "multiply", "params": {"by": "9"}
 },
 {
  "type" : "condition", "params": {"operator": "eq", "value": "42"}
 }
...
```

I don't think this will work will it?

Reply:
Yes, that would still work. We're comparing the value in the `result` field, which the multiply adapter would take its input from result, multiply the value, and update the output result value as the input for the next task.